spec/setfrozentag_action_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::SetfrozentagAction do
+  let(:test_repository) { 'test-repository' }
+  let(:test_token) { 'Test-GithubToken-1234' }
+  let(:test_milestone) do
+    { title: '10.1', number: '1234' }
+  end
+  let(:client) do
+    instance_double(
+      Octokit::Client,
+      list_milestones: [test_milestone],
+      update_milestone: nil,
+      user: instance_double('User', name: 'test'),
+      'auto_paginate=': nil
+    )
+  end
+
+  before do
+    allow(Octokit::Client).to receive(:new).and_return(client)
+  end
+
+  it 'properly passes the environment variable `GITHUB_TOKEN` to Octokit::Client' do
+    ENV['GITHUB_TOKEN'] = test_token
+    expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+    run_action_without_key(:github_token)
+  end
+
+  it 'properly passes the parameter `:github_token` all the way to Octokit::Client' do
+    expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+    run_described_fastlane_action(default_params)
+  end
+
+  it 'prioritizes `:github_token` parameter over `GITHUB_TOKEN` environment variable if both are present' do
+    ENV['GITHUB_TOKEN'] = 'Test-EnvGithubToken-1234'
+    expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+    run_described_fastlane_action(default_params)
+  end
+
+  it 'raises an error when the milestone is not found or does not exist' do
+    allow(client).to receive(:list_milestones).and_return([])
+    expect { run_described_fastlane_action(default_params) }.to raise_error(FastlaneCore::Interface::FastlaneError, 'Milestone 10.1 not found.')
+  end
+
+  it 'freezes the milestone adding ❄️ to the title' do
+    expect(client).to receive(:update_milestone).with(test_repository, test_milestone[:number], title: '10.1 ❄️')
+    run_action_with(:freeze, true)
+  end
+
+  it 'does not freeze the milestone if is already frozen' do
+    allow(client).to receive(:list_milestones).and_return([{ title: '10.2 ❄️', number: '1234' }])
+    expect(client).not_to receive(:update_milestone)
+    run_action_with(:milestone, '10.2 ❄️')
+  end
+
+  it 'does not freeze the milestone if :freeze parameter is false' do
+    expect(client).to receive(:update_milestone).with(test_repository, test_milestone[:number], title: '10.1')
+    run_action_with(:freeze, false)
+  end
```

Comment:
Self-Review: This for me is an odd behavior of this action If the `:freeze` parameter is false, it does not add the ❄️ symbol to the milestone title, as expected. However, it keeps calling the `update_milestone` to do an "unnecessary" update at that milestone.

Shouldn't in this case, the `update_milestone` not be called, following the same logic in place when the milestone is already frozen?

Reply:
The call is not unnecessary, as it will remove any existing ❄️  emoji from a frozen milestone title if it contains one — i.e. update milestone `20.9 ❄️ ` to `20.9`. This is used when we finalize a release at the end of a sprint, just before closing the milestone (and freezing the next one on the next code freeze).

Reply:
Yeah, I didn't notice that behavior. 🤔 
I add a test on it in any case, about removing any existing ❄️ emoji from a frozen milestone. 😄